src/js/ripple/orderbook.js

```diff
@@ -439,19 +439,22 @@ OrderBook.prototype.setFundedAmount = function(offer, fundedAmount) {
     return offer;
   }
 
-  var isOfferGetsExceeded = Amount.from_json(fundedAmount)
+  var isOfferGetsExceeded = Amount.from_json(
+    this._currencyGets.is_native()
+    ? fundedAmount
+    : fundedAmount + OrderBook.IOU_SUFFIX
```

Comment:
It's more trouble than it's worth to set the issuer to something other than an IOU placeholder. I'd say that it's actually worthless, as it makes no difference and is never exposed.

Calling a function on the OrderBook prototype won't de-duplicate very much.